plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseAutomaticJoinPushdownTest.java

```diff
@@ -75,38 +75,38 @@ public void testJoinPushdownAutomatic()
     {
         Session session = joinPushdownAutomatic(getSession());
 
-        try (TestTable left = joinTestTable("left", 6_000, 750);
-                TestTable right = joinTestTable("right", 1_000, 1_000)) {
+        try (TestTable left = joinTestTable("left", 600, 75);
+                TestTable right = joinTestTable("right", 100, 100)) {
             gatherStats(left.getName());
             gatherStats(right.getName());
 
             String simpleJoinQuery = "SELECT * FROM %s l JOIN %s r ON l.key = r.key";
-            // estimated left table size is ~444_000 bytes
-            // estimated right table size is ~74_000 bytes
-            // estimated join size is ~834_000
+            // estimated left table size is ~44_400 bytes
+            // estimated right table size is ~7_400 bytes
+            // estimated join size is ~83_400
 
             // with default configuration such join should not be pushed down;
-            // allowed join_to_tables ratio is 1.25 hence join size need to be less than (444_000 + 74_000) * 1.25 == 647_500
+            // allowed join_to_tables ratio is 1.25 hence join size need to be less than (44_400 + 7_400) * 1.25 == 64_750
             assertThat(query(session, format(simpleJoinQuery, left.getName(), right.getName())))
                     .isNotFullyPushedDown(joinOverTableScans());
 
-            // relax allowed ratio to 2.0; base line is 834_000 / (444_000 + 74_000) == 1.61 but we add some margin to cover possible mistakes in NDV calculations.
-            assertThat(query(maxJoinToTablesRatio(session, 2.0), format(simpleJoinQuery, left.getName(), right.getName())))
+            // relax allowed ratio to 3.0; base line is 111_200 / (44_400 + 7_400) == 2.14 but we add some margin to cover possible mistakes in NDV calculations.
```

Comment:
where does 111_200 come from? the estimated join size was 83_400 AFAIK.

Reply:
That's what I would expect, too, after consistently decreasing data size 10x, but actually its 111300 (`joinDataSize` when I put the breakpoint in https://github.com/trinodb/trino/blob/095d35adac6697ebdf5ed17b710e4e8a3eb27265/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcJoinPushdownUtil.java#L110), so I changed the numbers to match the reality and relaxed the ration even more.